### PR TITLE
flavors rename to instance types and preference added

### PR DESCRIPTION
**Pull request by RoniKishner**
##### Short description:
- Flavors CR name was changed to instance type
- Added preference CR

##### Which issue(s) this PR fixes:
new CR added for CNV 4.12 release

##### Special notes for reviewer:
Flavors constant in resource.py is kept until 4.11 is officially out since it is still being used by older versions, it should be deleted in the future.

### CODE REVIEW

**Comment by RoniKishner** [comment]
!verified

File: ocp_resources/resource.py [hunk 1]
```diff
@@ -313,7 +313,9 @@ class ApiGroup:
         DATA_IMPORT_CRON_TEMPLATE_KUBEVIRT_IO = "dataimportcrontemplate.kubevirt.io"
         EVENTS_K8S_IO = "events.k8s.io"
         FORKLIFT_KONVEYOR_IO = "forklift.konveyor.io"
+        # TODO flavor renamed to instancetype from 4.12, should be removed in the future
         FLAVOR_KUBEVIRT_IO = "flavor.kubevirt.io"
+        INSTANCE_TYPE_KUBEVIRT_IO = "instancetype.kubevirt.io"
```
Comment (myakove) [inline ocp_resources/resource.py:318]: Is this only for ocp 4.12?

Comment (RoniKishner) [reply ocp_resources/resource.py:318]: Yes the naming change from 4.12

**Comment by RoniKishner** [comment]
!verified

**Review by myakove — APPROVED** [review]

**Pull request merged**